### PR TITLE
Remove Jaeger Exporter - part 1: remove references in docs

### LINKED ISSUE #4638: Remove JaegerExporter
The spec has already been updated to remove JaegerExporter. https://github.com/open-telemetry/opentelemetry-specification/issues/3551

We need to remove JaegerExporter project and any references to use JaegerExporter:
- [x] [OpenTelemetry.Exporter.Jaeger](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.Jaeger)
- [x] [OpenTelemetry.Exporter.Jaeger.Tests](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/test/OpenTelemetry.Exporter.Jaeger.Tests)
- [x] [JaegerExporterBenchmarks.cs](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/Benchmarks/Exporter/JaegerExporterBenchmarks.cs)
- [x] [Examples](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/examples)
- [x] Main [README](https://github.com/open-telemetry/opentelemetry-dotnet#readme)
- [x] [VERSIONING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/VERSIONING.md)
- [x] README files under [docs/trace/](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace)

### PR DESCRIPTION
**Pull request by xiang17**
Fixes #4638
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

Remove Jaeger Exporter references in all docs.

Following PRs are to come which will remove Jaeger Exporter in examples, tests, and implementation.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated: N/A
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable): N/A

### CODE REVIEW

File: src/OpenTelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@
   `OTEL_DOTNET_EXPERIMENTAL_METRICS_EMIT_OVERFLOW_ATTRIBUTE` to `true` before
   setting up the `MeterProvider`.
   ([#4737](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4737))
+* Remove Jaeger Exporter from the core SDK. Jaeger supports OTLP protocol natively.
```
Comment (cijothomas) [inline src/OpenTelemetry/CHANGELOG.md:15]: Jaeger was never part of core sdk..

Comment (Kielek) [inline src/OpenTelemetry/CHANGELOG.md:15]:
```suggestion
* Remove Jaeger Exporter from the core components. Jaeger supports OTLP protocol natively.
```

I think that this message should go https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Jaeger/CHANGELOG.md or better you can put there comment similar to https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.MassTransit/CHANGELOG.md and https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.MassTransit/README.md while all code.

File: docs/trace/customizing-the-sdk/README.md [hunk 2]
```diff
@@ -620,21 +620,16 @@ components.
 Options classes can always be configured through code but users typically want to
 control key settings through configuration.
 
-The following example shows how to configure `JaegerExporterOptions` by binding
+The following example shows how to configure `ZipkinExporterOptions` by binding
```
Comment (reyang) [inline docs/trace/customizing-the-sdk/README.md:623]: Not a blocker, I think we want to encourage OTLP by default?

File: src/OpenTelemetry/CHANGELOG.md (hunk 1 above)
Comment (reyang) [reply src/OpenTelemetry/CHANGELOG.md:15]: I guess we don't need changelog updates. JaegerExporter's changelog will be removed as part of the exporter removal.

Comment (cijothomas) [reply src/OpenTelemetry/CHANGELOG.md:15]: It cannot be in the jaeger readme, as that is the thing we'll be deleting 🤣 (in subsequent PRs)
The nuget readme of jaeger can document its being discontinued.

File: docs/trace/customizing-the-sdk/README.md [hunk 3]
```diff
@@ -268,15 +268,15 @@ var tracerProvider = Sdk.CreateTracerProviderBuilder()
...
+[ZipkinExporter](../../../src/OpenTelemetry.Exporter.Zipkin/README.md) to the
 provider before it is built.
 
  ```csharp
 using OpenTelemetry;
 using OpenTelemetry.Trace;
 
 var tracerProvider = Sdk.CreateTracerProviderBuilder()
-    .AddJaegerExporter()
+    .AddZipkinExporter()
```
Comment (utpilla) [inline docs/trace/customizing-the-sdk/README.md:279]: Same as this https://github.com/open-telemetry/opentelemetry-dotnet/pull/4777/files#r1297351915. We could replace this with OTLP Exporter.

File: src/OpenTelemetry/CHANGELOG.md (hunk 1 above)
Comment (xiang17) [reply src/OpenTelemetry/CHANGELOG.md]: Removed the CHANGELOG entry.

Comment (xiang17) [reply src/OpenTelemetry/CHANGELOG.md]: Removed the line.

How do we change the nuget readme? Should we have an issue to keep track of the update?

File: docs/trace/customizing-the-sdk/README.md (hunk 2 above)
Comment (xiang17) [reply docs/trace/customizing-the-sdk/README.md]: Updated to OTLP exporter.

File: docs/trace/customizing-the-sdk/README.md (hunk 3 above)
Comment (xiang17) [reply docs/trace/customizing-the-sdk/README.md]: Updated it.

**Review by reyang — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

**Pull request merged**